haystack/preview/components/preprocessors/text_language_classifier.py

```diff
@@ -0,0 +1,82 @@
+import logging
+from typing import List, Dict, Any, Optional
+
+from haystack.preview import component, default_from_dict, default_to_dict
+from haystack.preview.lazy_imports import LazyImport
+
+logger = logging.getLogger(__name__)
+
+with LazyImport("Run 'pip install langdetect'") as langdetect_import:
+    import langdetect
+
+
+@component
+class TextLanguageClassifier:
+    """
+    Routes a text input onto one of different output connections depending on its language.
+    This is useful for routing queries to different models in a pipeline depending on their language.
+    The set of supported languages can be specified.
+    For routing Documents based on their language use the related DocumentLanguageClassifier component.
+
+    Example usage in a retrieval pipeline that passes only English language queries to the retriever:
+    document_store = MemoryDocumentStore()
```

Comment:
Nice example!.

I don't know if we need to prepend these examples with some special characters like ```

(@dfokina)

Reply:
Will do with ```python ... ``` 👍